rascaline/src/calculator.rs

```diff
@@ -245,15 +257,30 @@ impl Calculator {
         &self.parameters
     }
 
-    /// Get the set of keys this calculator would produce for the given systems
-    pub fn default_keys(&self, systems: &mut [Box<dyn System>]) -> Result<Labels, Error> {
-        self.implementation.keys(systems)
-    }
 
     #[time_graph::instrument(name="Calculator::prepare")]
     fn prepare(&mut self, systems: &mut [Box<dyn System>], options: CalculationOptions,) -> Result<TensorMap, Error> {
         // TODO: allow selecting a subset of keys?
-        let keys = self.implementation.keys(systems)?;
+        let default_keys=self.implementation.keys(systems)?;
+        let keys = match options.selected_keys {
+            Some(keys) => {
+                if default_keys.names() == keys.names() && !keys.is_empty() {
+                    keys.clone()
+                } else {
+                    if default_keys.names() != keys.names() {
+                        return Err(Error::InvalidParameter(format!(
+                            "The key names of the calculator [{}] and the passed keys [{}] do not match:",
+                            default_keys.names().join(", "),
+                            keys.names().join(", "))
+                            ));
+                    }
+                    else {
+                        return Err(Error::InvalidParameter(format!("Empty selected keys")));
+                    }
+                }
+            },
+            None => default_keys,
```

Comment:
This can be made a bit clearer with 

```suggestion
            Some(keys) if keys.is_empty() => {
                return Err(Error::InvalidParameter(format!("Empty selected keys")));
            }
            Some(keys) => {
                if default_keys.names() == keys.names() {
                    keys.clone()
                } else {
                    return Err(Error::InvalidParameter(format!(
                        "The key names of the calculator [{}] and the passed keys [{}] do not match:",
                        default_keys.names().join(", "),
                        keys.names().join(", "))
                    ));
                }
            }
            None => default_keys,
```